mouse-look.ts

```diff
@@ -38,71 +40,115 @@ export class MouseLookComponent extends Component {
     private rotationX = 0;
     private rotationY = 0;
     private mouseDown = false;
+    /** Pointerlock spec prevents calling pointerlock right after user exiting it via esc for ~1 second */
+    private pointerLockCooldown = false;
 
     onActivate() {
-        document.addEventListener('mousemove', this.onMouseMove);
-
         const canvas = this.engine.canvas;
+        if (this.mouseButtonIndex === 2) {
+            canvas.addEventListener('contextmenu', preventDefault, false);
+        }
+        canvas.addEventListener('pointermove', this.onPointerMove);
         if (this.pointerLockOnClick) {
-            canvas.addEventListener('mousedown', this.requestPointerLock);
+            canvas.addEventListener('pointerdown', this.onPointerDown);
+            document.addEventListener('pointerlockchange', this.onPointerLockChange);
         }
 
-        if (this.requireMouseDown) {
-            if (this.mouseButtonIndex === 2) {
-                canvas.addEventListener('contextmenu', preventDefault, false);
-            }
-            canvas.addEventListener('mousedown', this.onMouseDown);
-            canvas.addEventListener('mouseup', this.onMouseUp);
+        if (this.requireMouseDown && !this.pointerLockOnClick) {
+            canvas.addEventListener('pointerdown', this.onPointerDown);
+            canvas.addEventListener('pointerup', this.onPointerUp);
         }
     }
 
     onDeactivate() {
-        document.removeEventListener('mousemove', this.onMouseMove);
-
         const canvas = this.engine.canvas;
+        canvas.removeEventListener('pointermove', this.onPointerMove);
+
         if (this.pointerLockOnClick) {
-            canvas.removeEventListener('mousedown', this.requestPointerLock);
+            canvas.removeEventListener('pointerdown', this.onPointerDown);
```

Comment:
Pointerlock only really makes sense in a mouse on desktop context, though, right? Then having touch input request pointer lock isn't really desired.

Reply:
****